lib/bulkrax.rb

```diff
@@ -168,4 +169,43 @@ def self.multi_value_element_split_on
   def self.setup
     yield self
   end
+
+  # Responsible for stripping hidden characters from the given string.
+  #
+  # @param value [#to_s]
+  # @return [String] with hidden characters removed
+  #
+  # @see https://github.com/samvera-labs/bulkrax/issues/688
+  def self.normalize_string(value)
+    # Removing [Byte Order Mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark)
+    value.to_s.delete("\xEF\xBB\xBF")
+  end
+
+  # This class confirms to the Active::Support.serialze interface.  It's job is to ensure that we
+  # don't have keys with the tricksy Byte Order Mark character.
+  #
+  # @see https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html#method-i-serialize
+  class NormalizedJson
+    def self.normalize_keys(hash)
+      return hash unless hash.respond_to?(:each_pair)
+      returning_value = {}
+      hash.each_pair do |key, value|
+        returning_value[Bulkrax.normalize_string(key)] = value
+      end
+      returning_value
+    end
+
+    # Serializes an attribute value to a string that will be stored in the database.
+    def self.dump(value)
+      JSON.dump(normalize_keys(value))
+    end
+
+    # Deserializes a string from the database to an attribute value.
+    # rubocop:disable Security/JSONLoad
+    def self.load(string)
+      normalize_keys(JSON.load(string))
+    end
+    # rubocop:enable Security/JSONLoad
+  end
```

Comment:
1. similar to the above, is this a string that contains an object?
2. also, where is this called from?

Reply:
See above.